src/main/drivers/video_max7456.c

```diff
@@ -0,0 +1,620 @@
+#define DISABLE_MAX7456       GPIO_SetBits(MAX7456_CS_GPIO,   MAX7456_CS_PIN)
+#define ENABLE_MAX7456        GPIO_ResetBits(MAX7456_CS_GPIO, MAX7456_CS_PIN)
+
+textScreen_t max7456Screen;
+max7456State_t max7456State;
+static const extiConfig_t *max7456LOSExtiConfig;
+static const extiConfig_t *max7456VSYNCExtiConfig;
+static const extiConfig_t *max7456HSYNCExtiConfig;
+
+#if 0
+// for factory max7456 font
+static const uint8_t max7456_defaultFont_asciiToFontMapping[] = {
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, //  0  -  15
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, //  16 -  31
+    0x00, 0x00, 0x48, 0x00, 0x00, 0x00, 0x00, 0x46, 0x3F, 0x40, 0x00, 0x00, 0x45, 0x49, 0x41, 0x47, //  32 -  47  " !"#$%&'()*+,-./"
+    0x0A, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x44, 0x43, 0x4A, 0x00, 0x4B, 0x42, //  48 -  63  "0123456789:;<=>?"
+    0x4C, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F, 0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17, 0x18, 0x19, //  64 -  79  "@ABCDEFGHIJKLMNO"
+    0x1A, 0x1B, 0x1C, 0x1D, 0x1E, 0x1F, 0x20, 0x21, 0x22, 0x23, 0x24, 0x00, 0x46, 0x00, 0x00, 0x00, //  80 -  95  "PQRSTUVWXYZ[\]^_"
+    0x00, 0x25, 0x26, 0x27, 0x28, 0x29, 0x2A, 0x2B, 0x2C, 0x2D, 0x2E, 0x2F, 0x30, 0x31, 0x32, 0x33, //  96 - 111  "`abcdefghijklmno"
+    0x34, 0x35, 0x36, 0x37, 0x38, 0x39, 0x3A, 0x3B, 0x3C, 0x3D, 0x3E, 0x00, 0x00, 0x00, 0x00, 0x00, // 112 - 127  "pqrstuvwxyz{|}~ "
+};
+
+static const uint8_t max7456_defaultFont_fontToASCIIMapping[] = {
+    ' ', '1', '2', '3', '4', '5', '6', '7', '8', '9', '0', 'A', 'B', 'C', 'D', 'E', // 0x00 - 0x0F
+    'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', // 0x10 - 0x1F
+    'V', 'W', 'X', 'Y', 'Z', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', // 0x20 - 0x2F
+    'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', '(', // 0x30 - 0x3F
+    ')', '.', '?', ';', ':', ',', '\'', '/', '"', '-', '<','>', '@'                 // 0x40 - 0x4F
+};
+#endif
+
+
+static bool max7456ExtiHandler(const extiConfig_t *extiConfig)
+{
+    if (EXTI_GetITStatus(extiConfig->exti_line) == RESET) {
+        return false;
+    }
+
+    EXTI_ClearITPendingBit(extiConfig->exti_line);
+
+    return true;
+}
+
+
+void max7456_updateLOSState(void)
+{
+    // "LOS goes high when the VIN sync pulse is lost for 32 consecutive lines. LOS goes low when 32 consecutive valid sync pulses are received."
+    uint8_t status = GPIO_ReadInputDataBit(max7456LOSExtiConfig->gpioPort, max7456LOSExtiConfig->gpioPin);
+
+    max7456State.los = status != 0;
+
+    //debug[0] = max7456State.los;
+}
+
+void LOS_EXTI_Handler(void)
+{
+    static uint32_t callCount = 0;
+    bool set = max7456ExtiHandler(max7456LOSExtiConfig);
+    callCount++;
+    if (!set) {
+        return;
+    }
+
+    max7456State.losCounter++;
+
+    max7456_updateLOSState();
+}
+
+void VSYNC_EXTI_Handler(void)
```

Comment:
Ok I understand your meaning now.  I you typed in VSYNC twice in your first comment instead of VYSNC then HSYNC :smile: 

I think we'll go the simplest route for now.